src/arrow2/src/io/parquet/read/deserialize/nested_utils.rs

```diff
@@ -355,10 +355,11 @@ pub(super) fn extend<'a, D: NestedDecoder<'a>>(
     init: &[InitNested],
     items: &mut VecDeque<(NestedState, D::DecodedState)>,
     dict: Option<&'a D::Dictionary>,
-    remaining: &mut usize,
+    remaining: (&mut usize, &mut i64),
```

Comment:
Let's keep both these types as `usize`?

Reply:
Sounds good, change `num_values` to `usize` at all places